### ISSUE TITLE
divide by 0 for Int64 is inconsistent with int64

### ISSUE THREAD

**Issue by hhuuggoo** [pandas-dev/pandas#22793]
```
In [5]: pd.Series([1], dtype='Int64') / pd.Series([0], dtype='Int64')                                                                                                                                                                                                           
Out[5]:                                                                                                                                                                                                                                                                         
0   NaN                                                                                                                                                                                                                                                                         
dtype: float64     
```

Where as

```                                                                                                                                                                                                                                                 
In [6]: pd.Series([1], dtype='int64') / pd.Series([0], dtype='int64')                                                                                                                                                                                                           
Out[6]:                                                                                                                                                                                                                                                                         
0    inf                                                                                                                                                                                                                                                                        
dtype: float64        
```

It's because of this line:

https://github.com/pandas-dev/pandas/blob/0480f4c183a95712cb8ceaf5682c5b8dd02e0f21/pandas/core/arrays/integer.py#L545

Is it intentional for the new extension integer type to have different behavior here?

**Comment by WillAyd** [comment]
These should align though I think the extension type actually handles this more correctly

**Comment by jorisvandenbossche** [comment]
I am not sure we should change the behaviour of Series with default numpy types, as we then would deviate from numpy behaviour. 

Of course for the extension Int type, we have better handling of NA compared to numpy int, so which could be a reason to actually use that here. But, since division will always result in a floating dtype (and never integer), that's not a direct benefit in this case.

We have a `pd.options.mode.use_inf_as_na` so people can make this in practice to behave as NA, but that is False by default.

**Comment by jbrockmendel** [comment]
closed by #30183
